include/inexor/vulkan-renderer/wrapper/device.hpp

```diff
@@ -96,24 +96,65 @@ class Device {
         return m_transfer_queue_family_index;
     }
 
+    /// @brief Assign an internal Vulkan debug marker name to a Vulkan object.
+    /// This internal name can be seen in external debuggers like RenderDoc.
+    /// @note This method is only available in debug mode with ``VK_EXT_debug_marker`` device extension enabled.
+    /// @tparam Type The type of the Vulkan object.
+    /// @param object The Vulkan object.
+    /// @param object_type The Vulkan debug report object type.
+    /// @param name The internal name of the Vulkan object.
+    template <typename Type>
```

Comment:
I'm not sure if this function is worth templating anymore. Templating doesn't really provide any type safety and increases binary size a lot.

Reply:
Done.